### PR TITLE
Ember 2.16

### PR DESCRIPTION
**Pull request by jakesjews**
My company is stuck on ember-data 2.12 but I figured it is still good to keep this up to date. Upgraded to ember 2.16 and ran codemod for new module imports. Also remove old array.map polyfill.

### CODE REVIEW

**Comment by DingoEatingFuzz** [comment]
Looks good! It's wonderful that there is a codemod for modules.

**Comment by workmanw** [comment]
Can you give me one more day. I'm sorry for the delay. I have time tomorrow to review it.

**Comment by jakesjews** [comment]
@workmanw no rush at all :) I can't upgrade to this anyways

File: addon/array/stateful.js [hunk 1]
```diff
@@ -1,24 +1,23 @@
...
-
 /**
   A state-aware array that is tied to an attribute of a `DS.Model` instance.
 
   @class StatefulArray
   @namespace MF
   @extends Ember.ArrayProxy
 */
-const StatefulArray = Ember.ArrayProxy.extend(Ember.Copyable, {
+const StatefulArray = ArrayProxy.extend(Ember.Copyable, {
```
Comment (workmanw) [inline addon/array/stateful.js]: What about this `Ember.Copyable`? Is that one of the ones that's not exposed?

**Comment by workmanw** [comment]
Did you use the migration tool to do this work? Seeing it in the wild like this, I really like like the ES6 imports.

File: addon/array/stateful.js (hunk 1 above)
Comment (jakesjews) [reply addon/array/stateful.js]: Looks like its private https://emberjs.com/api/ember/2.16/classes/Ember.Copyable

**Comment by jakesjews** [comment]
@workmanw sorry about the late response. I used https://github.com/ember-cli/ember-modules-codemod for the conversion.

**Comment by workmanw** [comment]
Cool. Well this looks good to me. I'm also unable to test this in my application because of ember-data issues. But 👍 Thanks!

**Pull request merged**